src/checkout-buttons/strategies/apple-pay/apple-pay-button-strategy.ts

```diff
@@ -216,12 +216,23 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
         const { currency: { decimalPlaces } } = state.cart.getCartOrThrow();
         let checkout = state.checkout.getCheckoutOrThrow();
         const availableOptions = checkout.consignments[0].availableShippingOptions;
-        const shippingOptions = availableOptions?.map(option => ({
-            label: option.description,
-            amount: `${option.cost.toFixed(decimalPlaces)}`,
-            detail: option.additionalDescription,
-            identifier: option.id,
-        }));
+        const selectedOption = availableOptions?.find(({id}) => id === checkout.consignments[0].selectedShippingOption?.id);
+        const unselectedOptions = availableOptions?.filter(option => option.id !== checkout.consignments[0].selectedShippingOption?.id);
```

Comment:
Why not grab the `selectionShippingOptionId` after line 218 and that way the code will be much easier to read

Reply:
Hi @animesh1987, thanks for your review, I change this following your suggestion, could you please take another look?